### PR TITLE
Add tool version in patterns.json and retreive it in build.sbt

### PR DESCRIPTION
**Pull request by felitchia**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -24,6 +26,12 @@ enablePlugins(DockerPlugin)
 
 version in Docker := "1.0"
 
+val filename = "src/main/resources/docs/patterns.json"
```
Comment (pedrorijo91) [inline build.sbt]: private?

Comment (pedrorijo91) [inline build.sbt]: private?

File: build.sbt [hunk 2]
```diff
@@ -24,6 +26,12 @@ enablePlugins(DockerPlugin)
...
+
+val toolMap = JSON.parseFull(Source.fromFile(filename).getLines().mkString).get.asInstanceOf[Map[String,String]]
```
Comment (pedrorijo91) [inline build.sbt]: private?

File: build.sbt [hunk 3]
```diff
@@ -34,7 +42,7 @@ val installAll =
      |&& gem install activesupport
      |&& gem install parser:2.4.0.0
      |&& gem install pry
-     |&& gem install rubocop:0.49.0
+     |&& gem install rubocop:${rubocopVersion}
```
Comment (pedrorijo91) [inline build.sbt]: what happens if no version?

File: build.sbt (hunk 1 above)
Comment (felitchia) [inline build.sbt]: Does not work in a build.sbt file

Comment (pedrorijo91) [inline build.sbt]: oh, cute

File: build.sbt (hunk 3 above)
Comment (felitchia) [inline build.sbt]: Hm, since the idea was having the version in a unique place I left it implied that it would have to exist. But I'll add a default version for safety

File: build.sbt [hunk 4]
```diff
@@ -30,7 +30,11 @@ val filename = "src/main/resources/docs/patterns.json"
 
 val toolMap = JSON.parseFull(Source.fromFile(filename).getLines().mkString).get.asInstanceOf[Map[String,String]]
 
-val rubocopVersion = toolMap("version")
+val defaultVersion = "0.49.0"
+
+val rubocopVersion = { val jsonVersion = toolMap("version")
+  if(jsonVersion.isEmpty) defaultVersion else jsonVersion
```
Comment (pedrorijo91) [inline build.sbt]: when you have an option, the most idiomatic way to provide a default value would be by using `.map(...).getOrElse(...)` 

`val rubocopVersion = toolMap("version").getOrElse("0.49.0")`

File: build.sbt [hunk 5]
```diff
@@ -42,7 +46,7 @@ val installAll =
      |&& gem install activesupport
      |&& gem install parser:2.4.0.0
      |&& gem install pry
-     |&& gem install rubocop:${rubocopVersion}
+     |&& gem install rubocop:$rubocopVersion
```
Comment (pedrorijo91) [inline build.sbt]: Still, I'm not sure, but probably if you just run `gem install rubocop` it will install the latest rubocop version? Maybe that's the safest default value. This means my solution would be something like:

`gem install rubocop${rubocopVersion.map(version => s":$version").getOrElse("")}` (this probably doesn't work as it's written because of quotes and string interpolation chars nested)

Comment (felitchia) [inline build.sbt]: I'll check it

File: build.sbt (hunk 1 above)
Comment (machadoit) [inline build.sbt]: We shouldn't use the "src/main/..." hardcoded, there is a resourceDirectory for that. This should change to something like val jsonFile = (resourceDirectory in Compile).value / "docs" / "patterns.json".

On a related note, please don't call it "filename" there are a lot of files other than the patterns.json. patternsFile, jsonFile, something that give an hint.

**Comment by machadoit** [comment]
Sorry that I am late to the party @felitchia @pedrorijo91, but I really dislike the default version. That is a unexpected behaviour, because we can have a patterns.json that has patterns from one up-dated version and then it may fallback to one default that can be outdated for that patterns.json. Me no like it.

I think if the version is not present on the on the patterns.json we have to abort. I am no sbt expert, so I've checked how people use to "abort", and they usually throw an exception. If you want, you can check my suggestion of what I am talking about https://github.com/codacy/codacy-rubocop/commit/19f749aafa37be9093f9ff7af917cdab8b1ee273 you can do this solution or any other, just don't want a default version.

**TLDR**: Really don't care much about the implementation but:
 * Abort if the version is not present on the patterns.json
 * No hardcoded "src/main/resources"

**Comment by pedrorijo91** [comment]
LGTM. @machadoit will merge after finish fixing a script problem

**Comment by machadoit** [comment]
@pedrorijo91 already did the fix (just added the version to the patterns.json generation) if you wanna take another look, be free.

@mrfyda and @rtfpessoa wanna take a solution regarding the sbt task to retrieve the version or something? If not, for me this is ready to merge
